### PR TITLE
CLI 1.3.0, 0.29.1, 0.28.2 release notes

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: software/custom-image-registry.md [hunk 1]
```diff
@@ -167,7 +168,17 @@ Deploying code changes to a custom image registry requires triggering a GraphQL
 
 ## Push code to a custom registry
 
-After pushing images for your Astro project to your private registry, you can run a GraphQL query to push these images from your registry to Astronomer Software. At a minimum, your query has to include the following:
+You can use the Astro CLI to push build and push images from a Kubernetes cluster to your custom registry. Astronomer Software automatically recognizes that you pushed the image and pulls the image from the custom registry onto your Deployment.
```
Comment (jwitz) [inline software/custom-image-registry.md:171]:
```suggestion
You can use the Astro CLI to push build and push images from a Kubernetes cluster to your custom registry. Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry
```

File: software/custom-image-registry.md [hunk 2]
```diff
@@ -167,7 +168,17 @@ Deploying code changes to a custom image registry requires triggering a GraphQL
 
 ## Push code to a custom registry
 
-After pushing images for your Astro project to your private registry, you can run a GraphQL query to push these images from your registry to Astronomer Software. At a minimum, your query has to include the following:
+You can use the Astro CLI to push build and push images from a Kubernetes cluster to your custom registry. Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry
```
Comment (jwitz) [inline software/custom-image-registry.md:171]:
```suggestion
You can use the Astro CLI to push build and push images from a Kubernetes cluster to your custom registry. Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry.
```

File: astro/cli/release-notes.md [hunk 3]
```diff
@@ -9,6 +9,14 @@ This document provides a summary of all changes made to the [Astro CLI](cli/get-
 
 If you have any questions or a bug to report, reach out to [Astronomer support](https://support.astronomer.io).
 
+## Astro CLI v1.3.0
+
+Release date: July 12, 2022
+
+### Additional improvements
+
+- Upgraded the I to Go 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
```
Comment (kwcanuck) [inline astro/cli/release-notes.md:18]: Is _I to Go 1.18_ correct?

File: software/cli-release-notes.md [hunk 4]
```diff
@@ -31,6 +43,14 @@ You can now use `astro context list` and `astro context switch` to show the Astr
 
 These commands are intended for users who need to work across multiple Astronomer Software clusters or installations. They replace `astro cluster list` and `astro cluster switch`, respectively. For more information, see the [CLI Command Reference](cli-reference.md#astro-context-switch).
 
+## Astro CLI v0.29.1
+
+Release date: July 12, 2022
+
+### Bug fixes
+
+- Fixed an issue where `astro deploy` did not work when using Podman 4.0+
```
Comment (kwcanuck) [inline software/cli-release-notes.md:52]:
```suggestion
- Fixed an issue where `astro deploy` did not work when using Podman 4.0+.
```

File: software_versioned_docs/version-0.28/cli-release-notes.md [hunk 5]
```diff
@@ -41,6 +41,14 @@ You can now use `astro context list` and `astro context switch` to show the Astr
 
 These commands are intended for users who need to work across multiple Astronomer Software clusters or installations. They replace `astro cluster list` and `astro cluster switch`, respectively. For more information, see the [CLI command reference](cli-reference.md#astro-context-switch).
 
+## Astro CLI v0.28.2
+
+Release date: July 12, 2022
+
+### Bug fixes
+
+- Fixed an issue where `astro deploy` did not work when using Podman 4.0+
```
Comment (kwcanuck) [inline software_versioned_docs/version-0.28/cli-release-notes.md:50]:
```suggestion
- Fixed an issue where `astro deploy` did not work when using Podman 4.0+.
```

**Review by kwcanuck — APPROVED** [review]

**Review by sunkickr — APPROVED** [review]

**Comment by sunkickr** [comment]
@jwitz  we should test and update the custom image CI/CD with the new flag

File: astro/cli/astro-login.md [hunk 6]
```diff
@@ -13,6 +13,13 @@ Authenticate to Astro. After you run this command, the CLI prompts you for your
 astro login
 ```
 
+## Options
+
+| Option              | Description                                                                                   | Possible Values                                 |
+| ------------------- | --------------------------------------------------------------------------------------------- | ----------------------------------------------- |
+| `--l`, `--login-link` | Force the user to manually access the Cloud UI instead of opening it from the CLI           | ``|
+| `-t`, `--token-login` | Log in with a token from the Cloud UI                                        | ``    |
```
[the remark this answers is not in the document]
Comment (jwitz) [reply astro/cli/astro-login.md:21]: for `--login-link`? What's the exact workflow for logging in on a browserless machine?

File: astro/cli/astro-deploy.md [hunk 7]
```diff
@@ -21,16 +30,17 @@ astro deploy <options>
...
+| `-p`,`--prompt`           | Force the Deployment selection prompt even if a Deployment ID is specified                     | ``                                                     |
+| `--pytest`                | Deploy code to Astro only if the specified Pytests are passed                                  | ``                                                     |
+| `-s`,`--save`             | Save the current Deployment and working directory combination for future deploys               | ``                                                     |
+| `-t`,`--test`             | The filepath to an alternative pytest file or directory                                        | Valid filepath within your Astro project               |
+| `--workspace-id <string>` | In the prompt to select a Deployment, only show Deployments within this Workspace              | Any valid Workspace ID                                 |
+| `-i`, `--image-name`      | The name of a custom Runtime image to build your project with                                      | A valid name for a Docker image based on Astro Runtime |
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:43]: @sunkickr A few more questions about this flag:

- Do you have to pre-build the image before using this command? Does this process change at all? https://docs.astronomer.io/astro/develop-project#step-3-build-a-custom-docker-image
- Is the image name you specify here the same one in your Dockerfile? 
- Where does the custom image have to be located? In your Astro project?

File: astro/ci-cd.md [hunk 8]
```diff
@@ -220,7 +220,7 @@ To complete this setup, you need:
         - name: Deploy to Astro
           run: |
             curl -sSL install.astronomer.io | sudo bash -s
-            astro deploy
+            astro deploy --image-name "custom-<astro-runtime-image>"
```
Comment (jwitz) [inline astro/ci-cd.md:223]: @sunkickr I think this is the only update that needs to be made to the custom image template.

cc @stuart23 who wrote the original template- the CLI has a new flag that you can use to specify a custom image. Does this change any other part of the action? How does the CLI know where to look for the image?

File: astro/cli/astro-login.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (sunkickr) [reply astro/cli/astro-login.md:21]: --token-login. I know this is confusing but login-link still requires the machine to have a browser

File: software/cli-release-notes.md [hunk 9]
```diff
@@ -9,6 +9,18 @@ This document provides a summary of all changes made to the [Astro CLI](install-
...
+Release date: July 12, 2022
+
+### Deploy code to a custom image registry
+
+You can now use the Astro CLI to build and deploy an image to a [custom image registry](custom-image-registry.md). Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry.
+
+### Additional improvements
+
+- Upgraded the CLI to Go 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
+
```
Comment (neel-astro) [inline software/cli-release-notes.md:23]: There was another minor improvement added last minute, reference: https://github.com/astronomer/issues/issues/4844
Have made the `triggerer-replicas` flag's default value as `1` in `astro deployment create` for software

**Review by neel-astro — APPROVED** [review]
looks fine, there was a last-minute minor improvement added to the release, have mentioned the details in comment

File: astro/cli/astro-login.md [hunk 10]
```diff
@@ -13,6 +13,25 @@ Authenticate to Astro. After you run this command, the CLI prompts you for your
...
+
+## Examples
+
+```sh
+astro auth login
+# The CLI automatically opens the Cloud UI in a web browser
+astro auth login --login-link
+# The CLI provides a link to the Cloud UI that you can manually open in a web browser. You can
+astro auth login --token-login
+# The CLI provides a link to the Cloud UI that you can manually open in a web browser. You then copy a generated token from the UI and enter it in the CLI. For a browserless login, you can open the link and copy the token on a separate machine from the one running the Astro CLI.
```
Comment (paolaperaza) [inline astro/cli/astro-login.md:31]:
```suggestion
astro login
# The CLI automatically opens the Cloud UI in a web browser
astro login --login-link
# The CLI provides a link to the Cloud UI that you can manually open in a web browser. You can
astro login --token-login
# The CLI provides a link to the Cloud UI that you can manually open in a web browser. You then copy a generated token from the UI and enter it in the CLI. For a browserless login, you can open the link and copy the token on a separate machine from the one running the Astro CLI.
```

File: astro/ci-cd.md (hunk 8 above)
Comment (sunkickr) [reply astro/ci-cd.md:223]: @jwitz I think @stuart23 is making these changes in a separate PR. I do not think we should add this to this particular CI-CD pipelines example. Only the custom image pipeline needs to reference custom images

File: software_versioned_docs/version-0.28/cli-release-notes.md [hunk 11]
```diff
@@ -41,6 +41,14 @@ You can now use `astro context list` and `astro context switch` to show the Astr
 
 These commands are intended for users who need to work across multiple Astronomer Software clusters or installations. They replace `astro cluster list` and `astro cluster switch`, respectively. For more information, see the [CLI command reference](cli-reference.md#astro-context-switch).
 
+## Astro CLI v0.28.2
```
[the remark this answers is not in the document]
Comment (sunkickr) [reply software_versioned_docs/version-0.28/cli-release-notes.md:44]: customers using podman would want this version

File: astro/cli/release-notes.md [hunk 12]
```diff
@@ -9,11 +9,50 @@ This document provides a summary of all changes made to the [Astro CLI](cli/get-
...
+
+For more information about this command, see the [CLI command reference](cli/astro-deploy.md).
+
+### New token login method for Astro
+
+Astro CLI users can now log into Astro using an auth token by running `astro login --token-login`. If you run the command with this flag, the CLI prompts you to retrieve a token from the Cloud UI and enter it in the CLI.
+
+### Skip parsing DAGs before deploys
+
+You can configure the Astro CLI to automatically skip parsing DAGs before a deploy by updating one of the following configurations:
```
Comment (paolaperaza) [inline astro/cli/release-notes.md:37]:
```suggestion
By default, `astro deploy` automatically parses the DAGs in your Astro project for syntax and import errors. To develop more quickly, you can now configure the Astro CLI to automatically skip parsing DAGs before a deploy by updating one of the following configurations:
```

File: astro/cli/release-notes.md [hunk 13]
```diff
@@ -9,11 +9,50 @@ This document provides a summary of all changes made to the [Astro CLI](cli/get-
...
+
+- Add `skip_parse: true` to your `.astro/config.yaml` file.
+- Add `ASTRONOMER_SKIP_PARSE=true` as en environment variable to your local environment or CI/CD pipeline.
+
```
Comment (paolaperaza) [inline astro/cli/release-notes.md:41]:
```suggestion
- Add `ASTRONOMER_SKIP_PARSE=true` as en environment variable to your local environment or CI/CD pipeline.

For more information on parsing DAGs, see [Parse DAGs](test-and-troubleshoot-locally.md#parse-dags). For more information about deploying to Astro, see [Deploy code](deploy-code.md).
```

File: astro/cli/release-notes.md [hunk 14]
```diff
@@ -9,11 +9,50 @@ This document provides a summary of all changes made to the [Astro CLI](cli/get-
...
+### Additional improvements
+
+- Upgraded the CLI to Go 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
```
Comment (paolaperaza) [inline astro/cli/release-notes.md:44]:
```suggestion
- Upgraded the CLI to Go version 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
```

File: astro/cli/release-notes.md [hunk 15]
```diff
@@ -9,11 +9,50 @@ This document provides a summary of all changes made to the [Astro CLI](cli/get-
...
+- The CLI now provisions a triggerer when you create a Deployment using `astro deployment create`.
```
Comment (paolaperaza) [inline astro/cli/release-notes.md:45]:
```suggestion
```

This is software only im pretty sure actually -- we def always do triggerer on astro deployments? 🧐

File: software/cli-reference.md [hunk 16]
```diff
@@ -212,6 +212,7 @@ If you run `astro deploy` without specifying `your-deployment-release-name`, the
 | `--test`         | None       | A valid filepath within your Astro project to an alternative pytest file or directory. |
 | `--workspace-id` | String     | Lists available Deployments in your Workspace and prompts you to pick one.          |
 | `--no-cache`     | None       | Do not use any images from the container engine's cache when building your project. |
+| `-i`, `--image-name`      | The name of a pre-built custom Runtime image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```
Comment (paolaperaza) [inline software/cli-reference.md:215]:
```suggestion
| `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```

File: software/cli-reference.md [hunk 17]
```diff
@@ -614,6 +615,7 @@ Parse the DAGs in a locally hosted Astro project to quickly check them for error
 | Flag          | Value Type | Usage                                              |
 | ------------- | ---------- | -------------------------------------------------- |
 | `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |
+| `-i`, `--image-name`      | The name of a pre-built custom Runtime image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```
Comment (paolaperaza) [inline software/cli-reference.md:618]:
```suggestion
| `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```

File: software/cli-reference.md [hunk 18]
```diff
@@ -637,6 +639,7 @@ Run unit tests for your data pipelines with `pytest`, a testing framework for Py
 | ------------- | ---------- | -------------------------------------------------- |
 |`<pytest-filepath>`| String | Any valid filepath within the `tests` directory. |
 | `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |
+| `-i`, `--image-name`      | The name of a pre-built custom Runtime image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```
Comment (paolaperaza) [inline software/cli-reference.md:642]:
```suggestion
| `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```

File: software/cli-reference.md [hunk 19]
```diff
@@ -653,6 +656,7 @@ astro dev restart
 | Flag          | Value Type | Usage                                              |
 | ------------- | ---------- | -------------------------------------------------- |
 | `--env`       | string     | The filepath to your environment variables. The default is `.env`  |
+| `-i`, `--image-name`      | The name of a pre-built custom Runtime image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```
Comment (paolaperaza) [inline software/cli-reference.md:659]:
```suggestion
| `-i`, `--image-name`      | The name of a pre-built custom Docker image to use with your project. The image must be available from a Docker registry hosted on your local machine                                      | A valid name for a pre-built Docker image based on Astro Runtime |
```

File: software/cli-release-notes.md [hunk 20]
```diff
@@ -9,6 +9,19 @@ This document provides a summary of all changes made to the [Astro CLI](install-
 
 If you have any questions or a bug to report, contact [Astronomer Support](https://support.astronomer.io).
 
+## Astro CLI v1.3.0
+
+Release date: July 12, 2022
```
Comment (paolaperaza) [inline software/cli-release-notes.md:14]:
```suggestion
Release date: July 19, 2022
```

File: software/cli-release-notes.md [hunk 21]
```diff
@@ -9,6 +9,19 @@ This document provides a summary of all changes made to the [Astro CLI](install-
...
+
+### Deploy code to a custom image registry
+
+You can now use the Astro CLI to build and deploy an image to a [custom image registry](custom-image-registry.md). Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry.
```
Comment (paolaperaza) [inline software/cli-release-notes.md:18]:
```suggestion
You can now use the Astro CLI to build and deploy an image to a custom image registry. Based on the Helm configurations in your Kubernetes cluster, the Astro CLI automatically detects your custom image registry and pushes your image to it. It then calls the Houston API to update your Deployment to pull the new image from the registry.

For more information, see [Configure a custom registry for Deployment images](custom-image-registry.md).
```

File: software/cli-release-notes.md [hunk 22]
```diff
@@ -9,6 +9,19 @@ This document provides a summary of all changes made to the [Astro CLI](install-
...
+
+### Additional improvements
+
+- Upgraded the CLI to Go 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
```
Comment (paolaperaza) [inline software/cli-release-notes.md:22]:
```suggestion
- Upgraded the CLI to Go version 1.18, which includes improvements to both performance and the development experience. See the [Go Blog](https://go.dev/blog/go1.18).
```

File: software/custom-image-registry.md [hunk 23]
```diff
@@ -26,6 +26,7 @@ Deploying code changes to a custom image registry requires triggering a GraphQL
 
 - Helm.
 - kubectl.
+- Astro CLI version 1.0.3+.
```
Comment (paolaperaza) [inline software/custom-image-registry.md:29]:
```suggestion
- Astro CLI version 1.3.0+.
```

**Review by paolaperaza — APPROVED** [review]
@jwitz A few comments! Overall looks good but I did make some suggestions in release notes. I'll letchu finish off ✅

**Pull request merged**